### PR TITLE
CSCQVAIN-150 Change how path numbers are generialized in finding applying ui config

### PR DESCRIPTION
**Pull request by Kauhia**
Adds name descriptions for person names

### CODE REVIEW

File: src/schemas/fairdata-ida.ui.js [hunk 1]
```diff
@@ -202,6 +202,18 @@ export default {
 		'title': "Creator of the dataset",
 		'description': "The principal researcher or researchers involved in producing the data.",
 	},
+	'/properties/creator/*/oneOf/0/properties/name': {
```
Comment (jppiiroinen) [inline src/schemas/fairdata-ida.ui.js:205]: why there is `0` and why these changes to schema?

File: src/widgets/base.vue [hunk 2]
```diff
@@ -35,8 +35,6 @@ export default {
 		isValid() {
 			const statefromStore = this.$store.state.vState[this.path].v
 			return statefromStore !== null ? statefromStore : true;
-
-
 			//return (this.$store.state.vState[this.path].e || []).length === 0
```
Comment (jppiiroinen) [inline src/widgets/base.vue]: this commented out line could be removed at this time as well, it is not in use anyhow.

**Review by jppiiroinen — CHANGES_REQUESTED** [review]
The title is wrong. CSCVQVAIN-150 => CSCQVAIN-150.
Also the description in the title does not match what is written in the issue itself.
`Person name should always include instruction to write firstname lastname`

File: src/schemas/fairdata-ida.ui.js (hunk 1 above)
Comment (Kauhia) [reply src/schemas/fairdata-ida.ui.js:205]: Person objects only appear under oneOf relation where index 0 is Person and index 1 is Organization. These changes added the description defined in the issue.

Comment (jppiiroinen) [reply src/schemas/fairdata-ida.ui.js:205]: ok

File: src/store.js [hunk 3]
```diff
@@ -177,7 +177,20 @@ export default new Vuex.Store({
...
 		// uiForPath returns the UI overrides for the given path (if any)
 		uiForPath: (state) => (path) => {
-			return state.hints[path.replace(/(\/|^)[0-9]+(\/|$)/g, "$1*$2")] || {}
+			const searchPath = path
+				.split('/')
+				.filter(key => key !== '')
+				.map((key, index, array) => {
+					if (isNaN(key)) {
+						return key
+					} else if (array[index - 1] === 'oneOf') {
```
Comment (jppiiroinen) [inline src/store.js:186]: what if index is 0?
does it return undefined or the last item in array?

Comment (Kauhia) [reply src/store.js:186]: First key can never be a number so this should not be a problem. This is because there cannot be "indexed" content without something containing the "indexed" content. If this happens it should throw an index out of bounds error.
